scripts/plot.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+"""plot.py
+
+Usage:
+  plot.py -a ALGORITHM -g GEOMETRY -i INPUT... [-n] [-o OUTPUT_FILE]
+  plot.py (-h | --help)
+
+Options:
+  -h --help                         Show this screen.
+  -a ALGORITHM --algo ALGORITHM     Algorithm ['construction', 'knn_search', 'radius_search']
+  -g GEOMETRY --geometry GEOMETRY   Geometry (source cloud) ['filled_box', 'hollow_box']
+  -i FILE --input-files=FILE        Input file containing benchmark results
+  -o FILE --output-file=FILE        Output file
+  -n --numbers                      Print numbers
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+import re
+from docopt import docopt
+
+from tableau import tableau20
+
+from benchmark import parse_benchmark
+
+def libraries_comparison_rate_figure(input_file, algorithm, geometry, numbers):
+    # Phase 0: parse backends
+    backends = set()
+    with open(input_file) as f:
+        for line in f:
+            search_result = re.search("[^<]*<ArborX::BVH<([^>]*)>.*", line)
+            if search_result != None:
+                backends.add(search_result.group(1))
+    backends = list(backends)
+    backends.sort()
+
+    # Phase 1: data setup
+    num_sources  = []
+    timings = []
+    errors = []
+    unified_num_sources = [];
+    for i in range(len(backends)):
+      tmp_num_sources, _, tmp_timings = parse_benchmark(input_file, algorithm, 'ArborX::BVH<'+backends[i]+'>', 'median', geometry)
+      tmp_num_sources, _, tmp_errors = parse_benchmark(input_file, algorithm, 'ArborX::BVH<'+backends[i]+'>', 'stddev', geometry)
+      num_sources.append(tmp_num_sources)
+      timings.append(tmp_timings)
+      errors.append(tmp_errors)
+      unified_num_sources = unified_num_sources + tmp_num_sources
+
+    unified_num_sources = list(set(unified_num_sources))
+    unified_num_sources.sort()
+    num_source_labels = [None] * len(unified_num_sources)
+    for i in range(len(unified_num_sources)):
+        num_source_labels[i] = '$10^' + str(round(math.log(unified_num_sources[i], 10))) + '$'
+
+    # Transform given timings into the plotting data
+    data = np.zeros([len(backends), len(unified_num_sources)])
+    errors_data = np.zeros([len(backends), 2, len(unified_num_sources)])
+    for i in range(0, len(backends)):
+        for j in range(0, len(num_sources[i])):
+            index = unified_num_sources.index(num_sources[i][j])
+            data[i, index] = (num_sources[i][j] / timings[i][j])/10**6
+            errors_data[i, 0, index] = num_sources[i][j]/10**6*(1/timings[i][j] - 1/(timings[i][j]+errors[i][j]))
+            errors_data[i, 1, index] = num_sources[i][j]/10**6*(1/(timings[i][j]-errors[i][j]) - 1/timings[i][j])
```

Comment:
I think Damien is correct. Looking at the [propagation of uncertainty](https://en.wikipedia.org/wiki/Propagation_of_uncertainty#Example_formulae) for `f = A/B`, the deviation of the inverse `1/B` is indeed `dev(B)/B^2`. So, I think it would be sufficient to do this:
```suggestion
			scaling_factor = num_sources[i][j]/10**6
            data[i, index] = scaling_factor / timings[i][j]
            error = errors[i][j]/(timings[i][j]**2)
            errors_data[i, 0, index] = data[i, index] - scaling_factor * error
            errors_data[i, 1, index] = data[i, index] + scaling_factor * error
```
While it is still not exactly correct (as we are using medians), I think it should be closer. 

P.S. I have not tested this, and it may contain misprints.

Reply:
For me the question is how you want to interpret these bars and I normally think of them as some kind of confidence interval. I could be convinced that the original distribution is normal-like but the inverse is certainly not so I don't know how much value providing a standard deviation has.
In practice, the difference should be small anyway. 
If both of you rather want the formula above, tell me again and I will change.